src/stavenote.ts

```diff
@@ -1157,9 +1161,9 @@ export class StaveNote extends StemmableNote {
       const flagY =
         this.getStemDirection() === Stem.DOWN
           ? // Down stems are below the note head and have flags on the right.
-            y_top - noteStemHeight + 2
+            y_top - noteStemHeight + 2 - (this.flag !== undefined ? this.flag.checkMetrics().y_shift : 0)
```

Comment:
Can we separate this edit into two pieces?

```typescript
const extension = this.hasFlag() ? this.flag.checkMetrics().y_shift : 0;

this.flagY = ... 
  y_top - noteStemHeight + extension + 2
...
```

Reply:
done